### ISSUE TITLE
Static CDN Filters + fonts.googleapis.com results in "Cross-Origin Request Blocked"

### ISSUE THREAD

**Issue by raamdev** [wpsharks/comet-cache#487]
When Static CDN Filters are enabled on a site with a page that includes a web font like this (which may happen automatically when certain plugins/themes are installed):

``` html
<link rel='stylesheet' id='open-sans-css'  href='//fonts.googleapis.com/css?family=Open+Sans%3A300italic%2C400italic%2C600italic%2C300%2C400%2C600&#038;subset=latin%2Clatin-ext&#038;ver=4.2.2' type='text/css' media='all' />
```

The `//fonts.googleapis.com/css?family=Open+Sans...` portion results in the `Cross-Origin Request Blocked` error (see https://github.com/websharks/zencache/issues/427), and as a result anything on the site that used that font ends up broken.

In this particular case, to get around this issue, I needed to add `/css*` to the Static CDN Filters Blacklisted URI Exclusion Patterns.

---

ZenCache should, perhaps, detect some common domains where fonts are hosted (e.g., `fonts.googleapis.com`) and automatically exclude those from the Static CDN Filters.

**Comment by jaswrks** [comment]
> ZenCache should, perhaps, detect some common domains where fonts are hosted (e.g., fonts.googleapis.com) and automatically exclude those from the Static CDN Filters.

I see. It looks like this happens as a result of the HTML Compressor running first, because `fonts.googleapis.com` is not going to get picked up by the Static CDN Filters otherwise; i.e., `fonts.googleapis.com` is an external domain that cannot be served via the CDN unless the CSS itself is compressed/consolidated by the HTML Compressor and hosted locally.

So this seems like it could be a bug in the HTML Compressor actually. At least, it's a bug whenever running in concert with ZenCache Static CDN Filters. We might need to get creative to resolve this one.

**Comment by raamdev** [comment]
> this seems like it could be a bug in the HTML Compressor actually

Ah, that makes more sense. I was having trouble conceptually understanding why things were breaking with `fonts.googleapis.com` + Static CDN Filters. The HTML Compressor locally hosting the CSS + Static CDN Filters trying to serve said CSS via the CDN would explain why I was getting the "Cross-Origin Request Blocked".

**Comment by jaswrks** [comment]
@raamdev I'm closing the issue that I opened about this in the HTML Compressor repo, and giving it a label of `wontfix`. I added some thoughts here if you'd like to review and/or reopen. Please see: https://github.com/websharks/html-compressor/issues/60#issuecomment-112976893

---

_**Note:** This article at Google suggests that the issue you reported above may no longer exist. Please review: https://developers.google.com/fonts/docs/troubleshooting#CORS_header_

**Comment by raamdev** [comment]
> This article at Google suggests that the issue you reported above may no longer exist. Please review: https://developers.google.com/fonts/docs/troubleshooting#CORS_header

I just spent some time attempting to reproduce the issue I reported when I opened this issue and I'm not able to reproduce it. So either Google resolved the issue by sending the proper headers, or the site where I initially saw the issue occurring was using a CDN that was not sending the `Access-Control-Allow-Origin:*` header as it should.

I'm closing this. Thanks for the research and the comments in https://github.com/websharks/html-compressor/issues/60#issuecomment-112976893 @jaswsinc! :-)
